share/functions/change_number.fish

```diff
@@ -0,0 +1,31 @@
+function change_number
+    set -l cursor_position (commandline -C)
+    set -l current_buffer (commandline -b)
+    set -l count 0
+    set -l split_str (string split '' $current_buffer)
+    set -l char_position (math $cursor_position + 1)
+    set -l buffer_char $split_str[$char_position]
```

Comment:
This only handles a single digit. If you give it a number larger than that, it will add a "0" when it overflows.

E.g.

```
> echo 987
```

with the cursor on the "9" turns into `1087` because it adds 1 to the "9". And decreasing e.g. `echo 907` on the 0 turns it into `echo 9-17`.

I don't think that's how vim behaves in this case.

If you want, you can handle it, or I can do it.

Note: I did not start a review because the PR is still marked as "draft". If you're done, you should mark it as final.

Reply:
updated, btw could you add me to assignees thx